lib/galaxy/workflow/run_request.py

```diff
@@ -476,6 +502,20 @@ def add_parameter(name: str, value: str, type: WorkflowRequestInputParameter.typ
         workflow_invocation.step_states.append(step_state)
 
         if step.type == "subworkflow":
+            step.workflow_outputs
+            assert step.subworkflow
+            subworkflow: Workflow = step.subworkflow
+            effective_outputs: Optional[List[EffectiveOutput]] = None
+            if run_config.preferred_intermediate_object_store_id or run_config.preferred_outputs_object_store_id:
+                step_outputs = step.workflow_outputs
+                effective_outputs = []
+                for step_output in step_outputs:
+                    subworkflow_output = subworkflow.workflow_output_for(step_output.output_name)
+                    if subworkflow_output is not None:
+                        output_dict = EffectiveOutput(
+                            output_name=subworkflow_output.output_name, step_id=subworkflow_output.workflow_step_id
```

Comment:
the output_name and step_id don't necessarily make this unique if the same subworkflow is embedded more than once, is that going to be a problem ?

Reply:
I'll create an issue to write a test - I'm not sure off the top of my head.